arrows/ceres/tests/test_bundle_adjust.cxx

```diff
@@ -384,6 +384,21 @@ test_ba_intrinsic_sharing( camera_map_sptr cameras,
   return static_cast<unsigned int>(intrin_set.size());
 }
 
+// ----------------------------------------------------------------------------
+// Make sure each camera has unique (not shared) intrinsics
+camera_map_sptr make_intrinsics_unique(camera_map_sptr cameras)
+{
+  camera_map::map_camera_t new_cams;
+  for (auto ci : cameras->cameras())
+  {
+    auto cam = std::dynamic_pointer_cast<camera_perspective>(ci.second);
+    auto new_cam = std::make_shared<simple_camera_perspective>(
+      cam->center(), cam->rotation(), cam->intrinsics()->clone());
```

Comment:
need to check if cam==nullptr and complain if so instead of risking a core dump

Reply:
Not really a big deal here because this function is only exposed to unit tests.  The worst that could happen is that it causes a test failure if used on non-perspective cameras.  However, if someone copied this code to use elsewhere it could be a problem, so it's worth added the extra check.